lib/src/front_end/piece_factory.dart

```diff
@@ -896,7 +895,11 @@ mixin PieceFactory {
     //    var list = [
     //      element,
     //    ];
-    allowInnerSplit |= rightHandSide.canBlockSplit;
+    allowInnerSplit |= switch (rightHandSide) {
+      Expression() => rightHandSide.canBlockSplit,
+      DartPattern() => rightHandSide.canBlockSplit,
```

Comment:
Another option is to move `canBlockSplit` to be an extension on AstNode instead of separate extensions on Expression and DartPattern and then effectively move this type switch into there. But keeping it like this is fine too if you prefer it that way.

Reply:
That's a good point. I'm okay having them as separate extensions right now. But if it gets a little unwieldy, I'll make a CL to combine the two.